File: app/util/datetime/DateTimeAdapter.java
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package util.datetime;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.SerializerProvider;
import java.io.IOException;
import org.joda.time.DateTime;
import org.joda.time.format.ISODateTimeFormat;

public class DateTimeAdapter extends JsonSerializer<DateTime> {

    @Override
    public void serialize(DateTime value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
        gen.writeString(ISODateTimeFormat.dateTime().print(value));
    }
}

File: app/services/datetime/DateTimeAdapter.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package services.datetime

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.{JsonSerializer, SerializerProvider}
import org.joda.time.DateTime
import org.joda.time.format.ISODateTimeFormat

class DateTimeAdapter extends JsonSerializer[DateTime] {
  override def serialize(
      value: DateTime,
      gen: JsonGenerator,
      serializers: SerializerProvider
  ): Unit = {
    gen.writeString(ISODateTimeFormat.dateTime.print(value))
  }
}
